idmapping-rest/src/test/java/org/uniprot/api/idmapping/controller/IdMappingJobControllerIT.java

```diff
@@ -379,4 +382,91 @@ void jobSubmittedMoreThanAllowedIdsFailed() throws Exception {
                                                 + (this.maxCountCSV + 1)
                                                 + "' items.")));
     }
+
+    @Test
+    void validJobDetailsReturnDetails() throws Exception {
+        String jobId = "ID";
+        String ids = "Q1,Q2";
+        String taxId = "9606";
+        IdMappingJobRequest request = new IdMappingJobRequest();
+        request.setFrom(IdMappingFieldConfig.ACC_ID_STR);
+        request.setTo(IdMappingFieldConfig.GENE_NAME_STR);
+        request.setIds(ids);
+        request.setTaxId(taxId);
+
+        IdMappingJob job =
+                IdMappingJob.builder()
+                        .idMappingRequest(request)
+                        .jobStatus(JobStatus.NEW)
+                        .jobId(jobId)
+                        .build();
+        when(cacheService.getJobAsResource(jobId)).thenReturn(job);
+
+        // when
+        ResultActions response =
+                mockMvc.perform(
+                        get(JOB_DETAILS_ENDPOINT + "/" + jobId)
+                                .header(ACCEPT, MediaType.APPLICATION_JSON));
+        // then
+        response.andDo(log())
+                .andExpect(status().is(HttpStatus.OK.value()))
+                .andExpect(header().string(HttpHeaders.CONTENT_TYPE, APPLICATION_JSON_VALUE))
+                .andExpect(jsonPath("$.from", is(IdMappingFieldConfig.ACC_ID_STR)))
+                .andExpect(jsonPath("$.to", is(IdMappingFieldConfig.GENE_NAME_STR)))
+                .andExpect(jsonPath("$.ids", is(ids)))
+                .andExpect(jsonPath("$.taxId", is(taxId)))
+                .andExpect(jsonPath("$.redirectURL").doesNotExist());
+    }
+
+    @Test
+    void validJobDetailsStatusFinishedReturnDetailsWithRedirectURL() throws Exception {
+        String jobId = "ID";
+        String ids = "Q1,Q2";
+        String taxId = "9606";
+        IdMappingJobRequest request = new IdMappingJobRequest();
+        request.setFrom(IdMappingFieldConfig.ACC_ID_STR);
+        request.setTo(IdMappingFieldConfig.UNIPARC_STR);
+        request.setIds(ids);
+        request.setTaxId(taxId);
+
+        IdMappingJob job =
+                IdMappingJob.builder()
+                        .idMappingRequest(request)
+                        .jobStatus(JobStatus.FINISHED)
+                        .jobId(jobId)
+                        .build();
+        when(cacheService.getJobAsResource(jobId)).thenReturn(job);
+
+        // when
+        ResultActions response =
+                mockMvc.perform(
+                        get(JOB_DETAILS_ENDPOINT + "/" + jobId)
+                                .header(ACCEPT, MediaType.APPLICATION_JSON));
```

Comment:
Optional - If we use method MockMvcRequestBuilders.get(String urlTemplate, Object... uriVars), it will look cleaner.